pygsti/tools/matrixtools.py

```diff
@@ -1398,78 +1454,6 @@ def safe_dot(a, b):
         return _np.dot(a, b)
 
 
-def safe_real(a, inplace=False, check=False):
-    """
-    Get the real-part of `a`, where `a` can be either a dense array or a sparse matrix.
-
-    Parameters
-    ----------
-    a : numpy.ndarray or scipy.sparse matrix.
-        Array to take real part of.
-
-    inplace : bool, optional
-        Whether this operation should be done in-place.
-
-    check : bool, optional
-        If True, raise a `ValueError` if `a` has a nonzero imaginary part.
-
-    Returns
-    -------
-    numpy.ndarray or scipy.sparse matrix
-    """
-    if check:
-        assert(safe_norm(a, 'imag') < 1e-6), "Check failed: taking real-part of matrix w/nonzero imaginary part"
-    if _sps.issparse(a):
-        if _sps.isspmatrix_csr(a):
-            if inplace:
-                ret = _sps.csr_matrix((_np.real(a.data), a.indices, a.indptr), shape=a.shape, dtype='d')
-            else:  # copy
-                ret = _sps.csr_matrix((_np.real(a.data).copy(), a.indices.copy(),
-                                       a.indptr.copy()), shape=a.shape, dtype='d')
-            ret.eliminate_zeros()
-            return ret
-        else:
-            raise NotImplementedError("safe_real() doesn't work with %s matrices yet" % str(type(a)))
-    else:
-        return _np.real(a)
-
-
-def safe_imag(a, inplace=False, check=False):
-    """
-    Get the imaginary-part of `a`, where `a` can be either a dense array or a sparse matrix.
-
-    Parameters
-    ----------
-    a : numpy.ndarray or scipy.sparse matrix.
-        Array to take imaginary part of.
-
-    inplace : bool, optional
-        Whether this operation should be done in-place.
-
-    check : bool, optional
-        If True, raise a `ValueError` if `a` has a nonzero real part.
-
-    Returns
-    -------
-    numpy.ndarray or scipy.sparse matrix
-    """
-    if check:
-        assert(safe_norm(a, 'real') < 1e-6), "Check failed: taking imag-part of matrix w/nonzero real part"
-    if _sps.issparse(a):
-        if _sps.isspmatrix_csr(a):
-            if inplace:
-                ret = _sps.csr_matrix((_np.imag(a.data), a.indices, a.indptr), shape=a.shape, dtype='d')
-            else:  # copy
-                ret = _sps.csr_matrix((_np.imag(a.data).copy(), a.indices.copy(),
-                                       a.indptr.copy()), shape=a.shape, dtype='d')
-            ret.eliminate_zeros()
-            return ret
-        else:
-            raise NotImplementedError("safe_real() doesn't work with %s matrices yet" % str(type(a)))
-    else:
-        return _np.imag(a)
-
-
```

Comment:
I looked at where these functions were called in pygsti. The ``check`` was never set to True and sparse matrices were never passed. So I removed these functions and replaced with inlined code as needed.